src/tests/bulkConcepts/containers/BulkConceptsPage.test.js

```diff
@@ -333,6 +334,39 @@ describe('Test suite for BulkConceptsPage component', () => {
     expect(instance.componentDidUpdate(2)).toEqual(undefined);
   });
 
+  it('it should fetch filtered concepts when filter is true and current page is incremented', () => {
+    const props = {
+      setCurrentPage: jest.fn(),
+      currentPage: 1,
+      fetchBulkConcepts: jest.fn(),
+      filterConcept: jest.fn(),
+      concepts: [],
+      loading: true,
+      datatypes: [],
+      classes: [],
+      conceptLimit: 10,
+      match: {
+        params: {
+          type: 'users',
+          typeName: 'emasys',
+          collectionName: 'dev-org',
+          language: 'en',
+          dictionaryName: INTERNAL_MAPPING_DEFAULT_SOURCE,
+        },
+      },
+      addToFilterList: jest.fn(),
+      fetchFilteredConcepts: jest.fn(),
+      addConcept: jest.fn(),
+      previewConcept: jest.fn(),
+      handleNextPage: jest.fn(),
+    };
+    const wrapper = shallow(<BulkConceptsPage {...props} />);
+    const instance = wrapper.instance();
+    wrapper.setState({ filterOn: true });
+    instance.componentDidUpdate(props.currentPage + 1);
```

Comment:
Will this test fail if the above page is not incremented?

Reply:
No, it will not. What the test actually looks for is whether an update is made in the component and then uses the current page to fetch the concepts. Let me change the description so that it's not misleading in terms of what the test does.

Reply:
https://en.wikipedia.org/wiki/Test-driven_development

Reply:
I updated the test to remove the increment in the current page props.